java-generator/core/src/main/java/io/fabric8/java/generator/nodes/AbstractJSONSchema2Pojo.java

```diff
@@ -140,7 +140,7 @@ public static AbstractJSONSchema2Pojo fromJsonSchema(
       return fromJsonSchema.apply(JPrimitiveNameAndType.INT_OR_STRING);
     } else if (type == null
         && Boolean.TRUE.equals(prop.getXKubernetesPreserveUnknownFields())) {
-      return fromJsonSchema.apply(new JObjectNameAndType(key));
+      return fromJsonSchema.apply(JPrimitiveNameAndType.JSON_NODE);
```

Comment:
By doing this, we're using `JsonNode` abstraction as a part of the POJO, which might sound a little bit confusing. There's another use case in which we do this, partially, i.e. when encoding default values, but that is limited to the intermediate representation, rather than spanning to the actual final POJO. 
I understand this to be a better choice to be compliant with the spec, so basically I do agree. 
The only question left in my mind would actually be whether you've considered any other alternatives and what the rationale is for ending up with `JsonNode`.

Reply:
Great question @fabiobrz !

Let me try to answer thoroughly:
I have re-read the spec again and now I understand that in case of `type == null` and `x-kubernetes-preserve-unknown-fields` we should accept any value, including primitive types, arrays, maps and objects.

In Java this "can" be represented using an "object" and "JsonUnwrapped", but, IMHO this encoding is really awkward and has a number of downsides (e.g. it spills in the parent, depends on a specific Jackson functionality etc.)

Another alternative would be to map the field to `java.lang.Object` which gives us a "plain" Java construct without any guarantees.

The encoding with `JsonNode` gives us something that can be serialized and should conform to a valid Json which is all we can say about this field.
The downside is that we have little to no control and we leave everything in the hands of the final user, but, I can argue that there is almost nothing that can be done given the lack of guarantees.

Reply:
Thanks @andreaTP - I see what your reasoning is and I do agree on what you said about `JsonUnwrapped`.
For sure `Object` would offere less guarantees than `JsonNode`, although as said, it looks awkward as well how it mixes with the POJO representation, and this was my main concern.
We could let the change go in as-is, and agree on filing an issue for evaluating alternatives to improve things in the future, although it would break the APIs.

WDYT?